src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate lastDate = LocalDate.parse(dateTo, formatter);
+
+        StringBuilder result = new StringBuilder("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo);
+
+        for (String name: names) {
+            int totalSalary = 0;
+            result.append(System.lineSeparator()).append(name).append(" - ");
+
+            for (int i = 0; i < data.length; i++) {
+                String[] arrayElements = data[i].split(" ");
```

Comment:
this should be SEPARATOR constant

Reply:
I don't understand why I need this constant

Reply:
It makes code maintainable, let's imagine report format is changed and now client uses coma instead of space. With constant we open class, see constant SEPARATOR, changing it and we're fine, with hardcoded value we should recheck logic, be sure that we change proper symbol, especially this matters in situations when we may have several different " " used and half of them belongs to different logic. This task is small to show necessity but you never know how fast will some class grow, in a week it may be 10 times bigger and then not using constants from beginning will hurt